deploy/build-server/Dockerfile

```diff
@@ -0,0 +1,96 @@
+FROM ubuntu:20.04
+
+SHELL ["/bin/bash", "-c"]
+# To make it easier for build and release pipelines to run apt-get,
+# configure apt to not require confirmation (assume the -y argument by default)
+ENV DEBIAN_FRONTEND=noninteractive
+RUN echo "APT::Get::Assume-Yes \"true\";" > /etc/apt/apt.conf.d/90assumeyes
+
+# core dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    # azure devops agent
+    ca-certificates \
+    curl \
+    git \
+    jq \
+    iputils-ping \
+    libcurl4 \
+    libicu66 \
+    libssl1.0 \
+    libunwind8 \
+    netcat \
+    # ruby tests
+    freetds-bin \
+    # docker install
+    gnupg2 \
+    lsb-release \
+    # dotnet
+    apt-transport-https \
+    # brightbox
+    software-properties-common \
+    # dev build
+    zip \
+    && rm -rf /var/lib/apt/lists/*
+
+# install azure CLI
+RUN curl -LsS https://aka.ms/InstallAzureCLIDeb | bash \
+  && rm -rf /var/lib/apt/lists/*
+
+# install docker
+RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg2 --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+RUN echo \
+  "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+  $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+RUN apt-get update && apt-get install docker-ce docker-ce-cli containerd.io
+
+# Install nvm and current node runtimes
+ENV NVM_DIR /root/.nvm
+ENV NODE_VERSION 12.18.2
+
+RUN curl -sL https://raw.githubusercontent.com/creationix/nvm/v0.38.0/install.sh | bash \
+  && . $NVM_DIR/nvm.sh \
+  && nvm install $NODE_VERSION \
+  # angular spa still on node 10 until upgraded
+  && nvm install 10.14.1 \
+  && nvm alias default $NODE_VERSION
+
+# Install Ruby via brightbox
+RUN apt-add-repository ppa:brightbox/ruby-ng
+RUN apt-get update
+RUN apt-get install ruby2.6
+
+# Install dotnet
+RUN curl https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb --output packages-microsoft-prod.deb
+RUN dpkg -i packages-microsoft-prod.deb
+RUN rm packages-microsoft-prod.deb
+RUN apt-get update && apt-get install -y dotnet-sdk-5.0
+
+# Install azure functions core tools
+RUN curl https://packages.microsoft.com/keys/microsoft.asc | gpg2 --dearmor > microsoft.gpg
+RUN mv microsoft.gpg /etc/apt/trusted.gpg.d/microsoft.gpg
+RUN echo "deb [arch=amd64] https://packages.microsoft.com/repos/microsoft-ubuntu-$(lsb_release -cs)-prod $(lsb_release -cs) main" \
+  > /etc/apt/sources.list.d/dotnetdev.list
+RUN apt-get update && apt-get install azure-functions-core-tools-3
```

Comment:
Doesn't this automatically install dotnet as a dependency?

Reply:
no, it doesn't unfortunately.